### DISCUSSION TITLE
how to sample conversations while keeping audio annotation length constant

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#329] [Ideas]
With the new conversation sampler we can extract sections that have exchanges between key child and others, but these sections may be more or less long, which means we have little control over how much data will be annotated, and we cannot match this eg across children.

One option is to sample conversations with the sampler, and then sample among them:
- remove conversations shorter than X seconds, clip the ones that are longer than X seconds selecting the beginning
(but this biases for certain types of conversations, and we have more conversation beginnings than ends)

- keep short conversations, clip the longer ones randomly - with that rule you just mentioned
this still means that we may not have the same amount of audio for all children
plus, some of the conversations won't be such

**Comment by lucasgautheron** [comment]
What about using a fixed-length for all sampled conversations if this is very important to keep their duration under control? e.g. post-process the samples this way:

```python
segments = pd.read_csv('conversations.csv')
segments['midpoint'] = (segments['segment_offset']+segments['segment_onset'])/2
segments['segment_onset'] = segments['midpoint']-15*1000
segments['segment_offset'] = segments['midpoint']+15*1000
```

In this case the clips are centered around the middle of the conversation.

One more thing to mention: this issue remains theoretical as long as we do not know the real distribution of conversation durations. it would be nice to plot it for a few corpora.

Another comment: maybe it's just better to annotate _more_ until this is figured out, i.e. use a lower-limit for the clips lengths rather than an upper-limit, so that each clip is longer than e.g. 30 seconds. Then you can decide how to subsample in a smarter way at a later time, when you have a better understanding of how this might affect your analysis. Of course, this is not as efficient, but it is safer.

**Comment by lucasgautheron** [comment]
(Also, if fixed-length is crucial, why not use the HighVolubilitySampler instead?)

**Comment by lucasgautheron** [comment]
Here's a plot of the distribution of the duration of the 10 conversations with the most turns per session for different corpora.
orange = its, blue = vtc.
x-axis = in seconds.

[conversations.pdf](https://github.com/LAAC-LSCP/ChildProject/files/7608074/conversations.pdf)

**Reply by alecristia** [reply]
the conversations.pdf plot by @lucasgautheron in the comment above may be of interest to @camiscaff @marisacasillas @ BLLManitoba @orasanen @ebergelson: it shows the duration (in seconds) of conversations found by looking for the top 10 stretches in terms of adult-child & child-adult conversational turns

**Comment by orasanen** [comment]
This looks interesting. How'd you interpret the difference between vtc and LENA? Why are vtc estimates longer?

**Reply by lucasgautheron** [reply]
Maybe because the VTC has a higher recall? (if a vocalization fails to be detected, this might spuriously "end" a conversation, and we may expect that to happen more often with the LENA?)

**Reply by orasanen** [reply]
That would make sense.

I guess there can potentially be relatively complex effects of diarizer recall/precision characteristics on some measures conducted from daylong data. Would it be useful if we would be able to somehow relate diarizer error patterns to the measures that are derived from the data, at least understanding the direction that different measures are likely to be biased towards, or more ideally also get some (rough) error bounds for measures. For instance, knowing when we can safely state that "there was more male speech than female speech in the recording", given the known diarizer precision/recall on male and female input. Or when can we say that "the average length of a conversation was x turns", knowing whether it was from LENA/vtc and consisting of specific speaker classes, and recall/precision characteristics in general. I believe that having a handle on the uncertainty due to diarization (or other tool errors) would be quite useful, as most daylong analysis manuscripts I've seen mostly just assume that tool accuracy is sufficient and constant across classes/phenomena/corpora of interest, and then go forwards with the data analyses. 

I wonder if that could be done analytically, or do we just need a numerical simulator? But I guess those formulas/simulations would also need some assumptions about the statistical properties of the data?

**Reply by lucasgautheron** [reply]
I agree that this would be very useful. We have been working on something similar with @alecristia in order to evaluate the impact of speaker confusion (in the case of the VTC) on the correlation between the amount of vocs attributed to each speaker. The idea is that confusion between speakers would lead to spurious correlations, which we need to assess. We implemented a bayesian model fitted to human annotations for that, hoping that this would allow us to 1) simulate the impact of these confusion rates on correlation analyses 2)  infer the true correlations directly from automated annotations by performing some kind of calibration with this model.
However:

- The model is computationally expensive
- There's a number of potentially oversimplifying assumptions in our model (e.g. assuming all vocalizations are indepentently detected/classified as part of each class, or assuming that a real vocalization can lead to either 0 detected vocalization or 1 but not more, etc.)

If we manage to overcome these two issues, however, I guess the model may be able to achieve what you suggest too.

**Reply by alecristia** [reply]
just to add there’s a student with strong maths I’m  trying to recruit to continue along the lines of the approach Lucas described, and she seemed interested in a collaborative approach, so we could indeed look at effects on conversations defined like this.
